src/test/java/bgu/spl/mics/MessageBrokerTest.java

```diff
@@ -1,19 +1,105 @@
 package bgu.spl.mics;
 
+import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
+import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.junit.jupiter.api.Assertions.fail;
 
 public class MessageBrokerTest {
+    TestSubscriber s1;
+    TestSubscriber s2;
+    MessageBroker m;
+
     @BeforeEach
-    public void setUp(){
+    public void setUp() {
+        m = MessageBrokerImpl.getInstance();
+        s1 = new TestSubscriber("s1");
+        s2 = new TestSubscriber("s2");
+    }
+
+    @AfterEach
+    public void tearDown() {
+        m.unregister(s1);
+        m.unregister(s2);
+    }
 
+    @Test
+    public void testSingleton() {
+        MessageBroker otherInstance = MessageBrokerImpl.getInstance();
+        assertTrue(otherInstance == m, "different instances of singleton MessageBroker exist!");
+    }
+
+    @Test
+    public void testEventSubscribe() {
+        m.sendEvent(new IntEvent(0));
+        m.sendEvent(new IntEvent(1));
+        safeAwaitMessage(true);
+        assertTrue(s1.getNum() == 0, "Subscriber 1 didn't get the first called event");
+        assertTrue(s2.getNum() == 1, "Subscriber 2 didn't get the second called event");
+        m.sendEvent(new IntEvent(2));
+        safeAwaitMessage( false);
+        assertTrue(s1.getNum() == 2, "Subscriber 1 didn't appear 3rd in a 2 object round-robin");
```

Comment:
I would add a check for s2. to see it is still 1

Reply:
Done